### PR TITLE
Disable CGO and use pure Go Sqlite library

### PR DESCRIPTION
**Pull request by albertogdd**
## Description

[K8S-9898](https://dt-rnd.atlassian.net/browse/K8S-9898)

Disable CGO and use pure Go library for Sqlite

Build time _should_ be decreased with this change (maybe too less to notice)

image

## How can this be tested?

- Everything should work fine
- DB exists and its populated (see pic for commands to check this)
image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ ARG GO_BUILD_TAGS
 
 COPY pkg ./pkg
 COPY cmd ./cmd
-RUN --mount=type=cache,target="/root/.cache/go-build" CGO_ENABLED=1 CGO_CFLAGS="-O2 -Wno-return-local-addr" \
+RUN --mount=type=cache,target="/root/.cache/go-build" CGO_ENABLED=0 \
```
Comment (andriisoldatenko) [inline Dockerfile:17]:
```suggestion
RUN --mount=type=cache,target="/root/.cache/go-build" \
```
CGO_ENABLED=0 is default, you need to set it only when you want to enable it.

**Comment by albertogdd** [comment]
Indirect dependency seems to have an issue 
https://github.com/Dynatrace/dynatrace-operator/actions/runs/10163562292/job/28109526568#step:3:625

I will have a look after my moving days (Friday); feel free to commit or fix it in the meantime!

**Comment by andriisoldatenko** [comment]
> Should we merge this before the CSI rework?

yeah we agreed wtih @0sewa0 anad @chrismuellner that we can merge it.

**Comment by andriisoldatenko** [comment]
>Build time should be decreased with this change (maybe too less to notice)
you have to compare `main` branch witch CGO_ENABLED=1 and your branch:


```
➜  dynatrace-operator git:(main) time CGO_ENABLED=1 go build ./cmd/main.go
CGO_ENABLED=1 go build ./cmd/main.go  6.39s user 1.67s system 97% cpu 8.296 total
```

and 
```
➜  dynatrace-operator git:(feature/remove-cgo-and-use-pure-sqlite) time CGO_ENABLED=0 go build ./cmd/main.go
CGO_ENABLED=0 go build ./cmd/main.go  3.18s user 1.11s system 103% cpu 4.141 total
```
around ~ 2x times if you remove binary between builds and cache deps so no need to pull them.

**Review by waodim — COMMENTED** [review]
I took a look at the database it looks good to me, for example:

`sqlite> select * from code_modules;
dynatrace/dynatrace-codemodules:1.293.133.20240618-095559|/data/codemodules/ZHluYXRyYWNlL2R5bmF0cmFjZS1jb2RlbW9kdWxlczoxLjI5My4xMzMuMjAyNDA2MTgtMDk1NTU5|2024-07-31 09:30:52.509617554+00:00|2024-07-31 09:30:52.509617554+00:00|
sqlite> select * from tenant_configs;
0001-01-01 00:00:00+00:00|2024-07-31 10:38:04.768801415+00:00||ad866e25-72f0-4f6e-9115-40216b2645a4|dynakube-test|dynatrace/dynatrace-codemodules:1.293.133.20240618-095559|/data/asj34817/dynakube-test/config|asj34817|10
2024-07-31 09:20:36.096673575+00:00|2024-07-31 09:20:35.777074681+00:00||3d020f97-a9e3-4cf4-8836-35ffd86c1580|dynakube-test|custom-image|/data/asj34817/dynakube-test/config|asj34817|10`

If the CI runs through I would approve as the code also lgtm

**Pull request closed without merging**